### ISSUE TITLE
[1.16.4][Suggestion] Snow Real Magic compatibility?

### ISSUE THREAD

**Issue by The-Bernician-Lamb** [OreCruncher/DynamicSurroundings#665]
Hi,
This isn't really a bug, it's more of a suggestion, but I've been using a mod called Snow Real Magic lately that allows blocks to be "Snow-logged" so to speak. I've noticed that whenever I walk on these snow covered blocks it makes the sound for walking on planks. All that would be necessary is to make these blocks produce the sounds for walking on snow instead of the ones for walking on planks.

Below is an example of the "Snow-covered" blocks I'm talking about:
![2021-01-26_19 19 12](https://user-images.githubusercontent.com/64381954/106367878-c689b300-633d-11eb-8c43-1ad1b8c2f045.png)
As far as I know the mod has snow covered variants for all walls, slabs, stairs, fences, fence gates and some smaller vegetation like tall grass and flowers (Basically most non-full blocks).

Since the mod seems to generate these blocks dynamically based on what other mods are installed I'm not sure how easy this would be to accomplish, but like I said this is mainly a suggestion and isn't hugely important so take all the time you need for this.

Here is a link to the mod's page if you want to take a look at it: https://www.curseforge.com/minecraft/mc-mods/snow-real-magic

Update: I've made an issue about this on the Snow Real Magic Github to see what they can do to address this: https://github.com/Snownee/SnowRealMagic/issues/110

**Comment by OreCruncher** [comment]
Interesting mod.  I would have to take a look at it to see what is possible.

**Comment by The-Bernician-Lamb** [comment]
An update on this issue:

So, through the issue I linked in my previous comment that I posted on Snow Real Magic's Github I tried to see what they could do on their end to help with this issue, They were able to change the step sounds for their snow-covered blocks to sound like snow which is good!

However, Dynamic Surroundings seems to still process it as the sound for wood planks instead of the sound for snow, which is weird. So maybe the remaining issue is something to do with how your mod decides the step sounds for each block? I'm not a modder so feel free to correct me if I'm wrong.

**Comment by OreCruncher** [comment]
Yes.  DS will essentially replace the step sound processing of MC.  It sees these snow covered fences as regular fences.  I have to add some specific logic and/or configs to handle.  I haven't had a chance to look into it.
